scripts/methUnite.R

```diff
@@ -0,0 +1,217 @@
+  )
+}
+
+if (length(inputs) <= 1) {
+  stop("At least two samples required to perform merging")
+}
+
+
+# ---------------------------------------------------
+st <- system.time({
+
+  ## Hardlink Tabix files to avoid race conditions
+  inputFiles <- basename(inputs)
+  linkedFiles <- file.path(outdir, inputFiles)
+  file.link(inputs, linkedFiles)
+  file.link(paste0(inputs, ".tbi"), paste0(linkedFiles, ".tbi"))
+  inputs <- linkedFiles
+
+
+  if (length(inputs > 1)) {
+    inputs <- as.list(inputs)
+    sampleids <- as.list(sampleids)
+  }
+
+  ## Read data
+  message("Importing Tabix files.")
+  methylRawListDB <- methRead(
+    location = inputs,
+    sample.id = sampleids,
+    assembly = assembly,
+    dbtype = "tabix",
+    context = context,
+    treatment = treatments
+  )
+
+  ## Destrand if needed
+  if (destrand) {
+    message("Destranding first.")
+
+    destrandFun <- function(obj, cores) {
+      print(getSampleID(obj))
+
+      if (obj@resolution == "base") {
+        dir <- dirname(obj@dbpath)
+        filename <- paste(gsub(".txt.bgz", "", obj@dbpath),
+          "destrand.txt",
+          sep = "_"
+        )
+
+        filename <- basename(filename)
+
+        # need to use .CpG.dinuc.unifyOld because output needs to be ordered
+        newdbpath <- methylKit:::applyTbxByChr(obj@dbpath,
+          dir = dir, filename = filename,
+          return.type = "tabix",
+          FUN = function(x) {
+            options(scipen = 999)
+            methylKit:::.CpG.dinuc.unifyOld(methylKit:::.setMethylDBNames(
+              x,
+              "methylRawDB"
+            ))
+          },
+          mc.cores = cores
+        )
+
+        obj <- methylKit:::readMethylRawDB(
+          dbpath = newdbpath, dbtype = "tabix",
+          sample.id = obj@sample.id,
+          assembly = obj@assembly, context = obj@context,
+          resolution = obj@resolution
+        )
+      }
+      return(obj)
+    }
+
+    new.list <- suppressMessages(lapply(methylRawListDB, destrandFun, cores = cores))
+    methylRawListDB <- new("methylRawListDB", new.list, treatment = methylRawListDB@treatment)
+    destrandFiles <- getDBPath(methylRawListDB)
+  }
+
+  # remove output file if already exists
+  outFile <- file.path(outdir,sprintf("methylBase_%s.txt.bgz",suffix))
+  if( file.exists(outFile)) {
+    unlink(c(outFile, paste0(outFile, ".tbi")))
+  }
+  
+  ## Unite
+  message("Merging samples.")
+  methylBaseDB <- unite(methylRawListDB,
+    destrand = FALSE,
+    suffix = suffix,
+    dbdir = outdir,
+    mc.cores = cores,
+    chunk.size = 1e7,
+  )
+
+  ## FIXME: check wether result has more than 1 rows and fail if not
+
+  if (destrand) {
+
+    ## Remove temp destrand files
+    unlink(c(destrandFiles, paste0(destrandFiles, ".tbi")))
+  }
+
+  ## Remove temp hardlinked files
+  unlink(c(linkedFiles, paste0(linkedFiles, ".tbi")))
+})
+
+message("Done.")
+message("Process finished in (seconds): \n")
+print(st)
```

Comment:
You're doing this in all scripts, so I think it would be better to lift this machinery outside of the scripts and put it in the caller instead.